spec/project_spec.rb

```diff
@@ -109,11 +109,12 @@ class InvalidErrorHandler; end
     it 'should log an error when provided an invalid JSON datafile and skip_json_validation is true' do
       expect_any_instance_of(Optimizely::SimpleLogger).to receive(:log).once.with(Logger::ERROR, 'Provided datafile is in an invalid format.')
 
-      Optimizely::Project.new('{"foo": "bar"}', nil, nil, nil, true)
+      Optimizely::Project.new('{"foo": "bar"}', nil, nil, nil, false)
     end
 
     it 'should log an error when provided a datafile of unsupported version' do
-      expect_any_instance_of(Optimizely::SimpleLogger).to receive(:log).once.with(Logger::ERROR, 'Provided datafile is an unsupported version. Please use SDK version 1.1.2 or earlier for datafile version 1.')
+      config_body_invalid_json = JSON.parse(config_body_invalid_JSON)
```

Comment:
where config_body_invalid_json comes from?

Reply:
INVALID_CONFIG_BODY_JSON comes from spec_params.
we are parsing to get datafile version for log message.

INVALID_CONFIG_BODY = VALID_CONFIG_BODY.dup
INVALID_CONFIG_BODY['version'] = '5'
INVALID_CONFIG_BODY_JSON = JSON.dump(INVALID_CONFIG_BODY)